src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp

```diff
@@ -464,12 +478,14 @@ void SmokeCoAlarmServer::SendEvent(EndpointId endpointId, T & event)
 
 template <typename T>
 bool SmokeCoAlarmServer::GetAttribute(EndpointId endpointId, AttributeId attributeId,
-                                      EmberAfStatus (*getFn)(EndpointId endpointId, T * value), T & value) const
+                                      EmberAfStatus (*getFn)(EndpointId endpointId, T * value), T & value)
 {
     EmberAfStatus status = getFn(endpointId, &value);
     bool success         = (EMBER_ZCL_STATUS_SUCCESS == status);
+    mUnsupportedStatus   = (EMBER_ZCL_STATUS_UNSUPPORTED_ATTRIBUTE == status);
```

Comment:
See, this makes the methon non-const, because now we have this state it sets... Which is really mLastAttributeGetWasUnsupported or something....

It would _really_ be better if we had some sort of internal method with tri-state return, and the public API converts that to boolean, instead of smuggling things through a member like this.

Reply:
A simpler approach is to skip this step if GetAttribute() fails. Except for unsupported attribute errors, it is highly unlikely that SetAttribute() works correctly but other errors occur. And this change is equivalent to reverting to the pre #28751 state.